src/lib.rs

```diff
@@ -0,0 +1,235 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use discord_presence::Client;
+use rrplug::prelude::*;
+
+const EMPTY: String = String::new();
+
+static mut WAS_IN_GAME: bool = false;
+static mut RESET_SINGLE_PLAYER_TIMER: bool = true;
+
+struct ActivityData {
+    party: (u32, u32),
+    details: String,
+    state: String,
+    large_image: String,
+    large_text: String,
+    small_image: String,
+    small_text: String,
+    end: Option<u64>,
+    start: Option<u64>,
+}
+
+struct DiscordRpc {
+    gamestate: Option<GameState>,
+    serverinfo: Option<ServerInfo>,
+}
+
+impl Plugin for DiscordRpc {
+    fn new() -> Self {
+        Self {
+            gamestate: None,
+            serverinfo: None,
+        }
+    }
+
+    fn initialize(&mut self, external_plugin_data: ExternalPluginData) {
+        self.gamestate = external_plugin_data.get_game_state_struct();
+        self.serverinfo = external_plugin_data.get_server_info_struct();
+        println!("discord rpc plugin initialized");
+    }
+
+    fn main(&self) {
+        let gamestate = self.gamestate.as_ref().unwrap();
+        let serverinfo = self.serverinfo.as_ref().unwrap();
+
+        let mut drpc = Client::new(941428101429231617);
```

Comment:
What's with that hardcoded number? ^^

Reply:
its the app id.
https://github.com/R2Northstar/NorthstarDiscordRPC/blob/6fc14d7ad575fbea67dd76cca9cb46356ef903d0/DiscordRPC/discord_plugin.cpp#L106

Reply:
In that cause I'd do something like

```rust
let app_id = 941428101429231617;
let mut drpc = Client::new(app_id);
```

to avoid the case of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_\(programming\)) ^^